tests/python_tests/samples/conftest.py

```diff
@@ -0,0 +1,160 @@
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    }
+}
+
+TEST_FILES = {
+    "how_are_you_doing_today.wav": "https://storage.openvinotoolkit.org/models_contrib/speech/2021.2/librispeech_s5/how_are_you_doing_today.wav",
+    "adapter_model.safetensors": "https://huggingface.co/smangrul/tinyllama_lora_sql/resolve/main/adapter_model.safetensors"
+}
+
+SAMPLES_PY_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../samples/python"))
+SAMPLES_CPP_DIR = os.environ.get("SAMPLES_CPP_DIR", os.getcwd())
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--cleanup-cache", action="store_true", default=False, help="Cleanup cache after tests"
+    )
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_and_teardown(request, tmp_path_factory):
+    """Fixture to set up and tear down the temporary directories."""
+    
+    ov_cache = os.environ.get("OV_CACHE", tmp_path_factory.mktemp("ov_cache"))
+    models_dir = os.path.join(ov_cache, "test_models")
+    test_data = os.path.join(ov_cache, "test_data")
+    
+    logger.info(f"Creating directories: {models_dir} and {test_data}")
+    os.makedirs(models_dir, exist_ok=True)
+    os.makedirs(test_data, exist_ok=True)
+    
+    request.config.cache.set("OV_CACHE", str(ov_cache))
+    request.config.cache.set("MODELS_DIR", str(models_dir))
+    request.config.cache.set("TEST_DATA", str(test_data))
+    
+    yield
+    
+    if request.config.getoption("--cleanup-cache"):
+        if os.path.exists(ov_cache):
+            logger.info(f"Removing temporary directory: {ov_cache}")
+            shutil.rmtree(ov_cache)
+        else:
+            logger.info(f"Skipping cleanup of temporary directory: {ov_cache}")
+
+@pytest.fixture(scope="session")
+def convert_model(request):
+    """Fixture to convert the model once for the session."""
+    models_cache = request.config.cache.get("MODELS_DIR", None)
+    model_id = request.param
+    if model_id not in MODELS:
+        raise ValueError(f"Model ID '{model_id}' not found in MODELS dictionary.")
+    model_name = MODELS[model_id]["name"]
+    model_cache = os.path.join(models_cache, model_id)
+    model_path = os.path.join(model_cache, model_name)
+    model_args = MODELS[model_id].get("convert_args", [])
+    model_hf_cache = os.environ.get("HF_HOME", os.path.join(model_cache, "hf_cache"))
+    logger.info(f"Preparing model: {model_name}")
+    # Convert the model if not already converted
+    if not os.path.exists(model_path):
+        logger.info(f"Converting model: {model_name}")
+        command = [
+            "optimum-cli", "export", "openvino",
+            "--model", model_name, 
+            "--cache_dir", model_hf_cache, 
+            model_path
+        ]
+        if model_args:
+            command.extend(model_args)
+        logger.info(f"Conversion command: {' '.join(command)}")
+        retry_request(lambda: subprocess.run(command, check=True))
```

Comment:
I don't think `subprocess` would propagate the exception from `optimum-cli`. All the `subprocess` sees is the exit code.

Reply:
yes, I missed that, fixed by analyzing the error text